### PR TITLE
Create user Request

### PR DESCRIPTION
**Pull request by KristianGombosh**

### CODE REVIEW

File: src/main/java/com/afj/solution/buyitapp/controller/api/v1/UserController.java [hunk 1]
```diff
@@ -52,4 +61,20 @@ UserResponse me() {
         return userService.getMe(userId);
     }
 
+    @ApiOperation(value = "Create New User", authorizations = {@Authorization("Bearer")})
+    @ApiResponses({
+            @ApiResponse(code = 201, message = "User created"),
+            @ApiResponse(code = 409, message = "User Already exists"),
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/controller/api/v1/UserController.java:67]: `409` it is correct?

File: src/main/java/com/afj/solution/buyitapp/model/User.java [hunk 2]
```diff
@@ -140,4 +141,28 @@ public boolean isEnabled() {
...
+        final ZonedDateTime dateOfBirth = isNull(newUser.dateOfBirth) ? this.dateOfBirth : newUser.dateOfBirth;
+        final Set<GrantedAuthority> authorities = isNull(newUser.authorities) ? this.authorities : newUser.authorities;
+        currentUser.setFirstName(firstName);
+        currentUser.setLastName(lastName);
+        currentUser.setEmail(email);
+        currentUser.setUsername(username);
+        currentUser.setPhoneNumber(phoneNumber);
+        currentUser.setHomeAddress(homeAddress);
+        currentUser.setDateOfBirth(dateOfBirth);
+        currentUser.setAccountNonExpired(accountNonExpired);
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/model/User.java:161]: not need to update that field by `public User update(final User newUser, final User currentUser)` method.

File: src/main/java/com/afj/solution/buyitapp/model/User.java [hunk 3]
```diff
@@ -140,4 +141,28 @@ public boolean isEnabled() {
...
+        currentUser.setAccountNonLocked(accountNonLocked);
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/model/User.java:162]: not need to update that field by `public User update(final User newUser, final User currentUser)` method.

File: src/main/java/com/afj/solution/buyitapp/model/User.java [hunk 4]
```diff
@@ -140,4 +141,28 @@ public boolean isEnabled() {
...
+        currentUser.setCredentialsNonExpired(credentialsNonExpired);
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/model/User.java:163]: not need to update that field by `public User update(final User newUser, final User currentUser)` method.

File: src/main/java/com/afj/solution/buyitapp/model/User.java [hunk 5]
```diff
@@ -140,4 +141,28 @@ public boolean isEnabled() {
...
+        currentUser.setEnabled(enabled);
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/model/User.java:164]: not need to update that field by `public User update(final User newUser, final User currentUser)` method.

File: src/main/java/com/afj/solution/buyitapp/payload/request/CreateUserRequest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+
+    @ApiModelProperty(
+            name = "password",
+            dataType = "String",
+            value = "password",
+            example = "NewPassword!",
+            required = true
+    )
+    //@Password
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/payload/request/CreateUserRequest.java:76]: Why comment on it?

File: src/main/java/com/afj/solution/buyitapp/payload/request/CreateUserRequest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    private String password;
+
+    @ApiModelProperty(
+            name = "phoneNumber",
+            dataType = "String",
+            value = "User phone number",
+            example = "0975255207"
+    )
+    private String phoneNumber;
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/payload/request/CreateUserRequest.java:85]: Add validation to the Phone number

File: src/main/java/com/afj/solution/buyitapp/service/UserService.java [hunk 8]
```diff
@@ -15,4 +17,6 @@ public interface UserService {
     User saveAnonymous();
 
     UserResponse getMe(UUID userId);
+
+    Response<String> createUser(CreateUserRequest createUserRequest, UUID userId);
```
Comment (TomashGombosh) [inline src/main/java/com/afj/solution/buyitapp/service/UserService.java:21]: replace with -> `void createUser(CreateUserRequest createUserRequest, UUID userId);`

**Review by TomashGombosh — CHANGES_REQUESTED** [review]

File: src/main/java/com/afj/solution/buyitapp/controller/api/v1/UserController.java (hunk 1 above)
Comment (KristianGombosh) [reply src/main/java/com/afj/solution/buyitapp/controller/api/v1/UserController.java:67]: image

**Review by TomashGombosh — APPROVED** [review]

**Pull request merged**